src/rpc_client/src/lib.rs

```diff
@@ -159,3 +159,21 @@ macro_rules! rpc_client_method_impl {
         )*
     }
 }
+
+#[macro_export]
+macro_rules! meta_rpc_client_method_impl {
+    ($( { $client:tt, $fn_name:ident, $req:ty, $resp:ty }),*) => {
+        $(
+            pub async fn $fn_name(&self, request: $req) -> $crate::Result<$resp> {
+                let mut client = self.core.read().await.$client.to_owned();
+                match client.$fn_name(request).await {
+                    Ok(resp) => Ok(resp.into_inner()),
+                    Err(e) => {
+                        self.refresh_client_if_needed(e.code()).await;
```

Comment:
It seems `refresh_client_if_needed` doesn't deduplicate requests? i.e. if lots of requests receives error in a short time, lots of RPCs are sent too.

I've observed such case [here](https://buildkite.com/risingwavelabs/main-cron/builds/338#018638a4-7e58-45ee-bf19-ea045549f1d4): overwhelming `refresh election client failed could not refresh members` message.

Reply:
Yes, this is just an attempt to reconstruct the underlying channel, the retry is done by the upper level calls, as many calls cannot be retry. During the re-election interval all requests will fail, so a lot of logs will be generated, we can delete this log if it is too disturbing